dll/win32/browseui/shellbars/CBandSiteMenu.cpp

```diff
@@ -238,6 +253,9 @@ UINT CBandSiteMenu::_GetMenuIdFromISFBand(IUnknown *pBand)
     if (FAILED_UNEXPECTEDLY(hr))
         return UINT_MAX;
 
+    if (m_QLaunchPidl == NULL)
+        return UINT_MAX;
+
```

Comment:
I would prefer if you added a _GetQLaunchPidl function that would be called both here and in _CreateBuiltInISFBand. This function would return m_QLaunchPidl  if it is non null. If it is null it should call SHGetFolderPathAndSubDirW and initialize it once.

Reply:
Ok, added `_GetQLaunchPidl` function, and made `LPITEMIDLIST` variables initialization once.

But now, when I call Quick Launch bar second time (renaming the actual directory) it creates empty ISFBand without any warning. How do I fix this behaviour?